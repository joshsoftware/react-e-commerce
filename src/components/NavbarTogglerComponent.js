import React from 'react';
import { NavbarToggler } from 'reactstrap';
import PropTypes from 'prop-types';

const NavbarTogglerComponent = (props) => {
  const { toggle } = props;
  return <NavbarToggler onClick={toggle} />;
};

export default NavbarTogglerComponent;

NavbarTogglerComponent.propTypes = {
  toggle: PropTypes.func.isRequired
};
