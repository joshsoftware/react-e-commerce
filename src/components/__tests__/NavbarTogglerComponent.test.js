import NavbarTogglerComponent from '../NavbarTogglerComponent';
import { render } from '@testing-library/react';
import React from 'react';

describe('NavbarToggler Component', () => {
  it('Must return Dropdown Item', () => {
    const toggle = () => {};
    const { asFragment } = render(<NavbarTogglerComponent toggle={toggle} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
