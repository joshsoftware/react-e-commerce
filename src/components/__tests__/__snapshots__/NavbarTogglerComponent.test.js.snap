// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarToggler Component Must return Dropdown Item 1`] = `
<DocumentFragment>
  <button
    aria-label="Toggle navigation"
    class="navbar-toggler"
    type="button"
  >
    <span
      class="navbar-toggler-icon"
    />
  </button>
</DocumentFragment>
`;
